libs/community/langchain_community/llms/oci_data_science_model_deployment_endpoint.py

```diff
@@ -123,93 +264,293 @@ def _call(
-                data need to be sent to the endpoint.
-            endpoint (str):
-                The model HTTP endpoint.
-            header (dict, optional):
-                A dictionary of HTTP headers to send to the specified url.
-                Defaults to {}.
+            prompt (str):
+                The prompt to pass into the model.
+            stop (List[str], Optional):
+                List of stop words to use when generating.
             kwargs:
-                Additional ``**kwargs`` to pass to requests.post.
-        Raises:
-            Exception:
-                Raise when invoking fails.
+                requests_kwargs:
+                    Additional ``**kwargs`` to pass to requests.post
 
         Returns:
-            A JSON representation of a requests.Response object.
+            An iterator of GenerationChunks.
+
+
+        Example:
+
+            .. code-block:: python
+
+            async for chunk in oci_md.astream(("Tell me a joke."):
+                print(chunk, end="", flush=True)
+
         """
-        if not header:
-            header = {}
-        header["Content-Type"] = (
-            header.pop("content_type", DEFAULT_CONTENT_TYPE_JSON)
-            or DEFAULT_CONTENT_TYPE_JSON
-        )
-        request_kwargs = {"json": data}
-        request_kwargs["headers"] = header
-        timeout = kwargs.pop("timeout", DEFAULT_TIME_OUT)
-
-        attempts = 0
-        while attempts < 2:
-            request_kwargs["auth"] = self.auth.get("signer")
-            response = requests.post(
-                endpoint, timeout=timeout, **request_kwargs, **kwargs
-            )
-            if response.status_code == 401:
-                self._refresh_signer()
-                attempts += 1
-                continue
-            break
+        requests_kwargs = kwargs.pop("requests_kwargs", {})
+        self.streaming = True
+        params = self._invocation_params(stop, **kwargs)
+        body = self._construct_json_body(prompt, params)
 
-        try:
-            response.raise_for_status()
-            response_json = response.json()
+        async for line in await self.acompletion_with_retry(
+            data=body, run_manager=run_manager, stream=True, **requests_kwargs
+        ):
+            chunk = self._handle_sse_line(line)
+            if run_manager:
+                run_manager.on_llm_new_token(chunk.text, chunk=chunk)
+            yield chunk
 
-        except Exception:
-            logger.error(
-                "DEBUG INFO: request_kwargs=%s, status_code=%s, content=%s",
-                request_kwargs,
-                response.status_code,
-                response.content,
+    def _construct_json_body(self, prompt: str, params: dict) -> dict:
+        """Constructs the request body as a dictionary (JSON)."""
+        return {
+            "prompt": prompt,
+            **params,
+        }
+
+    def _invocation_params(self, stop: Optional[List[str]], **kwargs: Any) -> dict:
```

Comment:
`stop` is `Optional` but there is no default value at the method level here.

Reply:
fixed.